### PR TITLE
Fix contact addressBook switch

### LINKED ISSUE #174: contact is deleted after move to read only address book
### Steps to reproduce
1. User has an contact in his own address book
2. An other user share an address book with this user in read only mode
3. User move an contact to the shared address book

### Expected behaviour
1. the shared read only address book is not selectable or
2. you get an valid error message

### Actual behaviour
1. error message: "contact is not createable"
2. the origin contact is deleted

### Server configuration
**Operating system**:
rasperian jessy

**Web server:**
Apache

**Database:**
mysql 5.5.54

**PHP version:**
5.6.29

**Nextcloud version:** (see Nextcloud admin page)
11.0.2

**Contacts version:** (see Nextcloud apps page)
1.5.3

**Updated from an older Nextcloud or fresh install:**
updated
**Signing status:**
No errors have been found.

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "ocb41c453080",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/var\/local\/owncloud\/data",
        "dbtype": "mysql",
        "version": "11.0.2.7",
        "installed": true,
        "forcessl": false,
        "theme": "",
        "maintenance": false,
        "default_language": "de",
        "enable_previews": true,
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "owncloud",
        "mail_smtpmode": "smtp",
        "mail_domain": "cipicek.click",
        "mail_smtpsecure": "tls",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "posteo.de",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "loglevel": 1,
        "log_authfailip": true,
        "logfile": "\/var\/log\/owncloud\/owncloud.log",
        "log_rotate_size": 2097152,
        "logtimezone": "Europe\/Berlin",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trashbin_retention_obligation": "auto",
        "updatechecker": false,
        "appstoreenabled": true,
        "appstoreurl": "https:\/\/api.owncloud.com\/v1",
        "htaccess.RewriteBase": "\/",
        "appstore.experimental.enabled": true
    }
}

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
local

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**
firefox

**Operating system:**
debian strech

**CardDAV-clients:**
Web interface

### Logs
#### Web server error log
```
Insert your webserver log here
no log for this issue

#### Nextcloud log (data/nextcloud.log)
no log for this issue

### LINKED ISSUE #292: Contact disappear on addressbook switch
kazam_screencast_00000

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #292 

Testing: add contact, have at least 2 addressbooks and try changing the contact's addressbook
- The contact stays like it is
- The synchronisation with the server is successful
- After a page refresh, the contact has the addressbook you last set

### CODE REVIEW

**Comment by skjnldsv** [comment]
Also @DietmarMa and @dmp1ce, please test

**Comment by skjnldsv** [comment]
@irgendwie Anyway, this pr also change the ui switch problem where the contact just disappeared on addressbook switch, so I say, let's go for it and then fix the dav problem later :)

**Comment by skjnldsv** [comment]
Indeed, I edited the first part.

**Comment by skjnldsv** [comment]
Ping @Henni? @jancborchardt? @jonatoni :)

**Comment by skjnldsv** [comment]
Since it's a very little change, and irg agreed, merging.

**Comment by makuser** [comment]
> this pr also change the ui switch problem where the contact just disappeared on addressbook switch

What do you mean by that? When I move a contact to another AB, it disappears until I refresh the page. How would you recommend that I'd try your fix?

**Comment by skjnldsv** [comment]
@makuser When you moved a user from an ab to another, the contact simply disappeared until refresh.
It's not the case anymore :)

**Comment by makuser** [comment]
@skjnldsv Like I said: It still does. I did not understand what you mean, by "this pr also changes the ui switch problem", because in fact I still have this problem right now.

So, how would you recommend that I try to implement your fix on my server?

**Comment by skjnldsv** [comment]
@makuser wait for the next release! :)

**Comment by makuser** [comment]
@skjnldsv how can I test it anyway? I inserted the changes in the services.js but it doesn't get loaded on the client side, most likely due to the requested hash parameter.

You know OC/NC better than I do, so please give me a hint how to do it the right way.

We are currently adding hundreds of contacts to the database into six different address books and need to have it working properly.

**Comment by skjnldsv** [comment]
@makuser You can use our nightlies builds. See the readme!

**Comment by makuser** [comment]
@skjnldsv I have done that already, thanks for confirming this is the right way. Now the file gets loaded on the client side, but the issue still occurs sometimes.

If I move a contact to another addressbook, it sometimes disappears and another contact will be selected (and the uid parameter changed to this contact). If I then reload the page, the moved contact is in the list again and moved to the right destination AB.
So basically the same as before.

But at least it works in 4/5 tries. To debug the rest, I'd need more time. For now it's fine like it is.

Thanks guys, great work!
